### PR TITLE
Gradle build

### PR DESCRIPTION
**Pull request by ricellis**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant-labs/kafka-connect-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes - *build & test changes*
- [x] You have updated the [CHANGES.md](https://github.com/cloudant-labs/kafka-connect-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What

Changed build to use Gradle wrapper and added `Jenkinsfile`

## How

* Added Gradle wrapper and `build.gradle`.
* Updated tests to allow system property overrides of `test.properties` file.
* Cleaned up test code to simplify overrides, removed other code warnings (apologies for whitespace changes).
* Added `Jenkinsfile` with standard stages.
* Updated `.travis.yml` to use Gradle.
* Miscellaneous code tidy to avoid javadoc errors:
    * Fixed javadoc in `ResourceBundleUtil`
    * Removed unused `JsonUtil` class
    * Removed `InterruptedException` that was never used

## Testing

Tests pass in Jenkins.
Some tests fail in Travis, this appears to be a credentials issue that needs to be resolved outside of this PR.

## Issues

N/A

### CODE REVIEW

File: src/test/java/com/ibm/cloudant/kafka/connect/CloudantSourceTaskTest.java [hunk 1]
```diff
@@ -35,138 +21,132 @@
...
+                sourceProperties.get(InterfaceConst.PASSWORD),
+                data);
+
+        /*
+         * 2. Create connector
+         */
+        task = ConnectorUtils.createCloudantSourceConnector(sourceProperties);
+    }
+
+    private void expectOffsetLookupReturnNone() {
```
Comment (emlaver) [inline src/test/java/com/ibm/cloudant/kafka/connect/CloudantSourceTaskTest.java]: Remove?

File: README.md [hunk 2]
```diff
@@ -123,86 +123,6 @@ Junit tests are available in `src/test/java`. To execute, please modify values i
...
-```
-
-Release publish will publish maven artifacts to the maven staging repository.
-
-```
-Example:
-
-release.sh --release-publish --gitTag="v1.0.0"
+```sh
+./gradlew test
```
Comment (emlaver) [inline README.md]: Tests that fail for me locally:
```
com.ibm.cloudant.kafka.connect.CloudantSourceTaskTest > testStartWithIncrementalUpdates FAILED
    com.cloudant.client.org.lightcouch.CouchDbException at CloudantSourceTaskTest.java:105

com.ibm.cloudant.kafka.connect.CloudantSourceTaskTest > testStartMapOfStringString FAILED
    com.cloudant.client.org.lightcouch.CouchDbException at CloudantSourceTaskTest.java:76

com.ibm.cloudant.kafka.connect.CloudantSourceAndSinkTest > testReplicateAll FAILED
    com.cloudant.client.org.lightcouch.CouchDbException at CloudantSourceAndSinkTest.java:54

com.ibm.cloudant.kafka.connect.CloudantSinkTaskTest > testNonReplicateSinkRecordSchema FAILED
    com.cloudant.client.org.lightcouch.CouchDbException at CloudantSinkTaskTest.java:129

com.ibm.cloudant.kafka.connect.CloudantSinkTaskTest > testReplicateSinkRecordSchema FAILED
    com.cloudant.client.org.lightcouch.CouchDbException at CloudantSinkTaskTest.java:129
```

File: src/test/java/com/ibm/cloudant/kafka/connect/CloudantSourceTaskTest.java (hunk 1 above)
Comment (ricellis) [reply src/test/java/com/ibm/cloudant/kafka/connect/CloudantSourceTaskTest.java]: Done in c11f114da76c6e7033d41a8b3e7b36184867050b
Along with some other commented out code in the tests.

File: README.md (hunk 2 above)
Comment (ricellis) [reply README.md]: Those are the tests that actually use a remote server - did you point it to a Cloudant account with credentials via `test.properties`?

Perhaps the existing [README.md](https://github.com/cloudant-labs/kafka-connect-cloudant/blob/gradle-build/README.md#test-execution) message needs to be improved on this point - I've tried to make it a bit more clear in 646254b5bc7c2d812fb0344bb7787c4567aa77ac

Comment (emlaver) [reply README.md]: Ok, I missed the step to fill out `test.properties`.  All tests run successfully now.

**Pull request merged**